wasm/jsapi/table/set-reftypes.tentative.any.js

```diff
@@ -0,0 +1,40 @@
+// META: global=window,dedicatedworker,jsshell
+// META: script=assertions.js
+// META: script=/wasm/jsapi/wasm-constants.js
+// META: script=/wasm/jsapi/wasm-module-builder.js
+
+// Test cases for changes to the WebAssembly.Table.prototype.set() API that
+// come in with the reftypes proposal: the API makes the second argument optional and
+// if it is missing we should use DefaultValue of the table's element type.
+//
+// See:
+//   https://webassembly.github.io/reference-types/js-api/index.html#tables
+
+test(() => {
+    const builder = new WasmModuleBuilder();
+    builder
+      .addFunction("fn", kSig_v_v)
+      .addBody([])
+      .exportFunc();
+    const bin = builder.toBuffer();
+    const fn = new WebAssembly.Instance(new WebAssembly.Module(bin)).exports.fn;
+
+    const argument = { "element": "anyfunc", "initial": 1 };
+    const table = new WebAssembly.Table(argument, fn);
+    assert_equals(table.get(0), fn);
+
+    table.set(0);
+
+    assert_equals(table.get(0), null);
+  }, "Missing arguments for anyfunc table set");
+
+test(() => {
+  const testString = "foo";
+  const argument = { "element": "externref", "initial": 1 };
+  const table = new WebAssembly.Table(argument, testString);
+  assert_equals(table.get(0), testString);
+
+  table.set(0);
+
+  assert_equals(table.get(0), undefined);
+}, "Missing arguments for externref table set");
```

Comment:
I think we also need to test `set` with a `value` argument for at least externref and a primitive type